Recipes/definitions.recipe.json

```diff
@@ -3250,6 +3283,78 @@
               }
             }
           ]
+        },
+        {
+          "Name": "VideoModal",
+          "Settings": {},
+          "ContentPartFieldDefinitionRecords": [
+            {
+              "FieldName": "MediaField",
+              "Name": "ThumbnailImage",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Thumbnail Image",
+                  "Position": "0"
+                }
+              }
+            },
+            {
+              "FieldName": "TextField",
+              "Name": "ThumbnailVideo",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Thumbnail Video",
+                  "Position": "2"
+                },
+                "TextFieldSettings": {
+                  "Hint": "For example, a Vimeo embed URL with the background property. If populated, this will override the thumbnail image."
+                },
+                "ContentIndexSettings": {}
+              }
+            },
+            {
+              "FieldName": "LinkField",
+              "Name": "Video",
+              "Settings": {
+                "ContentPartFieldSettings": {
+                  "DisplayName": "Video",
+                  "Position": "3"
+                },
+                "LinkFieldSettings": {
+                  "Hint": "e.g a YouTube URL"
+                },
+                "ContentIndexSettings": {}
+              }
+            },
+            {
+              "FieldName": "CodeField",

```

Comment:
Instead of adding a separate field for this, we should use the `JavaScriptEventsPart`. That way if we add any other fields to `JavaScriptEventsPart`, this content type will automatically get it.

Reply:
I can't find any examples of JavaScriptEventsPart in the whole widgets definition, where is it defined?

Reply:
We've done this one before...

Reply:
If you don't remember - https://github.com/EtchUK/Etch.OrchardCore.Widgets/blob/main/Migrations.cs#L93